edx_api/enrollments/__init__.py

```diff
@@ -22,6 +25,88 @@ def __init__(self, requester, base_url):
         """
         self.requester = requester
         self.base_url = base_url
+        self._course_details = {}
+
+    def _get_course_details(self, course_id):
+        """
+        Retrieves course details.
+
+        Args:
+            course_id (str): Course id.
+        """
+        if course_id in self._course_details:
+            return self._course_details[course_id]
+
+        req_url = urljoin(self.base_url, self.course_details_url.format(course_key=course_id))
+        resp = self.requester.get(req_url)
+        resp.raise_for_status()
+        resp_json = resp.json()
+        self._course_details[course_id] = resp_json
+        return resp_json
+
+    def _parse_enrollments(self, enrollments_list):
+        """
+        Parse enrollments response into :class:`Enrollments`.
+        """
+        def populate_course_details(enrollment):
+            """Populates course details if missing."""
+            enrollment.pop('previous', None)
+            enrollment.pop('next', None)
+            if enrollment.get('course_details'):
+                return enrollment
+            course_details = self._get_course_details(enrollment['course_id'])
+            course_details['course_start'] = course_details['start']
+            course_details['course_end'] = course_details['end']
+            enrollment['course_details'] = course_details
+            return enrollment
+
+        enrollments = [populate_course_details(enrl) for enrl in enrollments_list]
+        return Enrollments(enrollments)
```

Comment:
This effectively discards all the data we just queried.
The current `Enrollments` model is designed to show course enrollments per user, so it sets up a dictionary where the key is the course_id, and the value is the `Enrollment` object.

https://github.com/mitodl/edx-api-client/blob/1978f388ae6567f741af0dab6025d5f16620e083/edx_api/enrollments/models.py#L20

That works for the previous calls, since no user will be enrolled in the same course twice.
When we load that with all users course enrollments however, it clobbers these and only stores one enrollment per course ID since it uses the same key for every enrollment with the course ID.

Reply:
Yes the current `Enrollments` class is not going to work for this. I wasn't sure how this is going to be used so I thought I made it work with the current model and then change it if necessary.
The only downside here is that querying the enrollments list API does not returns the course details while querying the user enrollment API does.